### PR TITLE
Import from anywhere

### LINKED ISSUE #593: [FEATURE] Imports from Anywhere on the FS
## User Story
1. As a {Jayvee user}
2. I want {to perform Imports from anywhere of my file system}
3. So that {model parsing and interpretation don't depend on the current directory (like in Python)}

## User Acceptance Criteria
- [ ] Imported documents are dynamically added to the LSP instead of being restricted to a fixed working directory.

See discussion in #590 for more details.

## Definitions of Done
- [ ] A PR has been opened and accepted
- [ ] All user acceptance criteria are met
- [ ] All tests are passing

### PR DESCRIPTION
**Pull request by georg-schwarz**
Closes #593

### CODE REVIEW

**Comment by georg-schwarz** [comment]
Ran in a roadblock in the implementation. The interpreter somehow doesn't pick up the hook added it seems. Asked for help in https://github.com/eclipse-langium/langium/discussions/1308#discussioncomment-10159256

**Review by rhazn — COMMENTED** [review]
While the code itself looks fine to me, I still get errors trying to run models with imports from above them if the interpreter runs from the same directory as the model:

```
error: Import from "./../../shared/transforms.jv" could be resolved. Check if the file exists in the given location.
$In <path removed>model.jv:15:8
15 | } from "./../../shared/transforms.jv";
```

File: libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
```
Comment (rhazn) [inline libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts:1]: It's 2024.

File: libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    assert(isJayveeModel(parsedModel), 'Test file is not valid Jayvee model');
+    return parsedModel;
+  }
+
+  beforeEach(() => {
+    // Create language services
+    services = createJayveeServices(NodeFileSystem).Jayvee;
+  });
+
+  const validCases: [string, string][] = [
```
Comment (rhazn) [inline libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts:47]: Nice :)

File: libs/language-server/src/lib/workspace/dynamic-file-import.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
```
Comment (rhazn) [inline libs/language-server/src/lib/workspace/dynamic-file-import.ts:1]: Same year thing here :D

File: libs/language-server/src/test/assets/import-dynamic-reference/invvalid-to-import-wrong-file-ending.jvm
Comment (rhazn) [inline libs/language-server/src/test/assets/import-dynamic-reference/invvalid-to-import-wrong-file-ending.jvm:1]: I think it is not good to make this file have both a typo (`invvalid`) and a wrong file ending (`jvm`). From the name I'd just keep the file ending incorrect.

**Review by rhazn — APPROVED** [review]
Looks good to me and works on my machine :).

File: libs/language-server/src/test/assets/import-dynamic-reference/invvalid-to-import-wrong-file-ending.jvm
Comment (georg-schwarz) [reply libs/language-server/src/test/assets/import-dynamic-reference/invvalid-to-import-wrong-file-ending.jvm:1]: Thx, was a unintended typo :)

File: libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts (hunk 1 above)
Comment (georg-schwarz) [reply libs/language-server/src/lib/lsp/import-dynamic-reference.spec.ts]: Done

**Pull request merged**